source/adios2/engine/bp3/BP3Writer.cpp

```diff
@@ -218,7 +218,8 @@ void BP3Writer::InitBPBuffer()
     else
     {
         m_BP3Serializer.PutProcessGroupIndex(
-            m_IO.m_Name, m_IO.m_HostLanguage,
+            m_IO.m_Name,
+            (m_IO.m_ArrayOrder == ArrayOrdering::RowMajor) ? "C++" : "Fortran",
```

Comment:
Since m_HostLanguage is still around under the IO class,  can we use it instead of deriving it from m_ArrayOrder?  I noticed this in a few places.

Reply:
Well, the idea is that the ArrayOrdering might not match the host language, so if they've change the array ordering in a C++ to be ColumnMajor (because that's how the app views the data) and then we write out using "C++", that ArrayOrdering specification will be effectively lost (because we don't have ArrayOrder directly in the BP3/4 header).  BP5 won't use Language as a stand-in for ArrayOrder, but in the old formats that it's in we use C++ as a standin for RowMajor and Fortran as a stand-in for ColumnMajor.

Reply:
Hmm this host language parameter, derived from array ordering, is later used to determine column/row major again in the serializers..  Might as well pass the arraying ordering directly to the serializers :)

Reply:
Yep.  For BP3/4 that the hostlanguage is used for that is a historical artifact.  We could rewrite them, but unless we change their file formats it's going to be converted to language specification before going to disk.  So, this is the minimalist approach to leaving those formats the same, but evolving so that new formats can use it directly.

Reply:
Understood. Thanks Greg